From ffb5bd17746a3ae4f67b2dcd6a4225683e15fb95 Mon Sep 17 00:00:00 2001
From: Thomas Lamprecht <thomas@lamprecht.org>
Date: Thu, 19 Sep 2024 09:25:09 +0200
Subject: [PATCH] sys: cherry-pick type-cast fix for foreign arch builds

This is the relevant part of upstream commit 2da31214 ("sys: crypt:
style + drop unnecessary length check")

Signed-off-by: Thomas Lamprecht <thomas@lamprecht.org>
---
 proxmox-sys/src/crypt.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/proxmox-sys/src/crypt.rs b/proxmox-sys/src/crypt.rs
index 3313f668..3248ecdc 100644
--- a/proxmox-sys/src/crypt.rs
+++ b/proxmox-sys/src/crypt.rs
@@ -71,7 +71,7 @@ pub fn crypt(password: &[u8], salt: &[u8]) -> Result<String, Error> {
         // > output field of their data argument, and crypt writes an invalid hash to its static
         // > storage area.  This string will be shorter than 13 characters, will begin with a ‘*’,
         // > and will not compare equal to setting.
-        if data.output.first().is_none() || Some(&('*' as i8)) == data.output.first() {
+        if data.output[0] == '*' as libc::c_char {
             bail!("internal error: crypt_r returned invalid hash");
         }
         CStr::from_ptr(&data.output as *const _)
@@ -133,7 +133,7 @@ pub fn crypt_gensalt(prefix: &str, count: u64, rbytes: &[u8]) -> Result<String,
     // while it states that this is "in addition" to returning a null pointer, this isn't how
     // `crypt_r` seems to behave (sometimes only setting an invalid hash) so add this here too just
     // in case.
-    if output.first().is_none() || Some(&('*' as i8)) == output.first() {
+    if output[0] == '*' as libc::c_char {
         bail!("internal error: crypt_gensalt_rn could not create a valid salt");
     }
 
